lib/mjsonwp/mjsonwp.js

```diff
@@ -124,10 +124,20 @@ function makeArgs (reqParams, jsonObj, payloadParams) {
       }
     }
   }
-  let args = _.flatten(realRequiredParams).map((p) => jsonObj[p]);
-  if (payloadParams.optional) {
-    args = args.concat(_.flatten(payloadParams.optional).map((p) => jsonObj[p]));
+  let args;
+  if (_.isFunction(payloadParams.makeArgs)) {
+    // if the spec defines a 'makeArgs' function, allow it to figure out how to
+    // turn the json inputs into a command arguments array
+    args = payloadParams.makeArgs(jsonObj);
```

Comment:
I would expect more detailed description of makeArgs property here - what it should do and what is the expected result/the type of it